clap_derive/src/utils/ty.rs

```diff
@@ -24,13 +26,21 @@ impl Ty {
 
         if is_unit_ty(ty) {
             t(Unit)
-        } else if is_generic_ty(ty, "Vec") {
-            t(Vec)
+        } else if let Some(subty) = subty_if_name(ty, "Vec") {
```

Comment:
This isn't really backwards compatible, because existing code could potentially already have a `Vec<Vec<T>>` as a value type. 

One possible option is to have use a newtype for a Vec that we define instead.... but that is kind of ugly IMO.